libs/payments/paypal/src/lib/paypal.service.ts

```diff
@@ -4,8 +4,39 @@
 
 import { Injectable } from '@nestjs/common';
 import { PayPalManager } from './paypal.manager';
+import { PaypalCustomerManager } from './paypalCustomer/paypalCustomer.manager';
+import { CurrencyManager } from '@fxa/payments/currency';
 
 @Injectable()
 export class PayPalService {
-  constructor(private paypalManager: PayPalManager) {}
+  constructor(
+    private paypalManager: PayPalManager,
+    private paypalCustomerManager: PaypalCustomerManager,
+    private currencyManager: CurrencyManager
+  ) {}
+
+  /**
+   * Create and verify a billing agreement is funded from the appropriate
+   * country given the currency of the billing agreement.
+   */
+  async createBillingAgreement(uid: string, token: string, currency: string) {
+    const billingAgreementId = await this.paypalManager.createBillingAgreement(
+      token
+    );
+
+    const agreementDetails = await this.paypalManager.getBillingAgreement(
+      billingAgreementId
+    );
+
+    const country = agreementDetails.countryCode;
+
+    this.currencyManager.assertCurrencyCompatibleWithCountry(currency, country);
```

Comment:
Would be good to get @bbangert's opinion on whether this is necessary anymore with the new plan structure. I'd reached out over slack and he'd mentioned perhaps no.